src/java/org/apache/cassandra/io/util/FileUtils.java

```diff
@@ -177,15 +178,24 @@ public static File createDeletableTempFile(String prefix, String suffix)
 
     public static Throwable deleteWithConfirm(String filePath, boolean expect, Throwable accumulate)
     {
-        return deleteWithConfirm(new File(filePath), expect, accumulate);
+        return deleteWithConfirm(new File(filePath), expect, accumulate, null);
     }
 
     public static Throwable deleteWithConfirm(File file, boolean expect, Throwable accumulate)
     {
+        return deleteWithConfirm(file, expect, accumulate, null);
+    }
+    
+    public static Throwable deleteWithConfirm(File file, boolean expect, Throwable accumulate, RateLimiter rateLimiter)
+    {
+        if (rateLimiter == null)
+            rateLimiter = RateLimiter.create(Double.MAX_VALUE);
+
         boolean exists = file.exists();
         assert exists || !expect : "attempted to delete non-existing file " + file.getName();
         try
         {
+            rateLimiter.acquire();
```

Comment:
Just skip `acquire()` call if `rateLimiter == null` instead of creating one conditionally?